src/Common/src/System/Diagnostics/Debug.cs

```diff
@@ -84,19 +125,31 @@ public static void WriteLine(string message)
         [System.Diagnostics.Conditional("DEBUG")]
         public static void Write(string message)
         {
-            s_logger.WriteCore(message ?? string.Empty);
+            if (message == null)
+            {
+                return;
+            }
+            if (NeedIndent)
+            {
+                WriteIndent();
+            }
+            WriteCore(message);
+            if (message.EndsWith(Environment.NewLine))
+            {
+                NeedIndent = true;
+            }
         }
 
         [System.Diagnostics.Conditional("DEBUG")]
         public static void WriteLine(object value)
         {
-            WriteLine((value == null) ? string.Empty : value.ToString());
+            WriteLine(value?.ToString() ?? string.Empty);
         }
 
         [System.Diagnostics.Conditional("DEBUG")]
         public static void WriteLine(object value, string category)
         {
-            WriteLine((value == null) ? string.Empty : value.ToString(), category);
+            WriteLine(value?.ToString() ?? string.Empty, category);
```

Comment:
Is the ```?? string.Empty``` necessary?  Seems like that'll be handled by the null check in the called method.